Add end2end tests to CI

berndbohmeier opened the pull request.

berndbohmeier committed:
Add end2end tests to CI

schmir-at-bb commented on .circleci/config.yml:
```diff
@@ -454,6 +454,25 @@ jobs:
+            cd tools/bridge
+            make test-end2end
+            cd ../..
+
```
This could be written as `make -C tools/bridge test-end2end`

berndbohmeier replied:
Ah okay, I copied it from the other circle ci code. I changed it everywhere.

berndbohmeier committed:
Use make -C to switch directories

schmir committed:
Get rid of the uptest_function argument

Besides being broken because the function is not called, it's also not
being used.

schmir committed:
Fix end2end tests on circle ci

It looks like parity is not mining transactions when running the
end2end tests on circle ci. Passing the --jsonrpc-server-threads=8
command line option seems to fix that.

These machines seem to have 36 CPUs. Maybe parity chooses the default
based on the number of CPUs and we run into a race condition with that
value.

schmir committed:
Enhance log message in end2end's Bridge.is_up
